test/test_memory.cpp

```diff
@@ -0,0 +1,79 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <regex>
+
+#include <fusion.h>
+#include <inlining.h>
+#include <ir/utils.h>
+#include <ops/alias.h>
+#include <ops/arith.h>
+#include <ops/utils.h>
+#include <options.h>
+#include <test/utils.h>
+#include <test/validator.h>
+#include <type.h>
+
+namespace nvfuser {
+
+class MemoryTest : public NVFuserTest {};
+
+TEST_F(MemoryTest, LoadCache) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeContigTensor(1);
+  fusion.addInput(tv0);
+  TensorView* tv1 =
+      ops::newValLike(tv0, tv0->getDataType().value())->as<TensorView>();
+  IrBuilder::create<LoadStoreOp>(
+      LoadStoreOpType::Set, tv1, tv0, CacheOp::AllLevels);
+  TensorView* tv2 = add(tv1, IrBuilder::create<Val>(1.0));
+  TensorView* tv3 = set(tv2);
+  fusion.addOutput(tv3);
+
+  tv1->split(0, 4);
+  tv1->split(0, 32);
+  TransformPropagatorWithCheck propagator(tv1);
+  MaxRootDomainInfoSpanningTree(tv1).traverse(&propagator);
+
+  // Parallelize LoadStoreOps. Other TensorViews don't support vectorization.
+  tv1->axis(0)->parallelize(ParallelType::BIDx);
+  tv1->axis(1)->parallelize(ParallelType::TIDx);
+  tv1->axis(2)->parallelize(ParallelType::Vectorize);
+  scheduler_utils::parallelizeAllLike(tv1, {tv3});
+
+  inlineMost();
+
+  at::Tensor input = at::randn(
+      {1024}, at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0));
+  at::Tensor expected_output = input + 1.0f;
+
+  FusionExecutor fe;
+  fe.setSaveCompiledBinaryFlag(true);
+  {
+    DebugDumpOptionsGuard debug_dump_options_guard;
+    DebugDumpOptionsGuard::getCurOptions().set(DebugDumpOption::Ptx);
+    fe.compileFusion(&fusion, {input});
+  }
+  std::vector<char> compiled_ptx = fe.compiledPtx();
```

Comment:
I'm having a hard time understanding how this is supposed to work. Since `comiple_to_sass` is `true` by default, is PTX string actually generated by just enabling the PTX debug dump option?

Reply:
It actually fails on my workstation.

```
[ RUN      ] MemoryTest.LoadCache
unknown file: Failure
C++ exception with description "!compiled_kernel.ptx.empty() INTERNAL ASSERT FAILED at "/home/nmaruyama/nvfuser/debug2/csrc/executor_utils.cpp":1149, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues.
Exception raised from compileSource at /home/nmaruyama/nvfuser/debug2/csrc/executor_utils.cpp:1149 (most recent call first):
```

My system setup:

```
 nvidia-smi                                                                                                                                                         ~/nvfuser/debug2
Mon Sep 11 14:38:39 2023
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 520.61.05    Driver Version: 520.61.05    CUDA Version: 11.8     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA TITAN RTX    On   | 00000000:65:00.0 Off |                  N/A |
| 41%   32C    P8    15W / 280W |      0MiB / 24576MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

Reply:
Sorry, I merged in the wrong way. It should be fixed now.